src/main/java/io/wispforest/owo/util/RecipeSpecificRemainderStorage.java

```diff
@@ -0,0 +1,58 @@
+package io.wispforest.owo.util;
+
+import io.wispforest.owo.mixin.recipe.RecipeMixin;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+
+import java.util.*;
+
+/**
+ *  Simple Class used to store and register all Recipe Specific Remainders.
+ */
+public final class RecipeSpecificRemainderStorage {
```

Comment:
This class should be renamed to `RecipeSpecificRemainders`. Additonally, for consistency, all methods in it which specify further than `Remainer` should be shortened: `registerRecipeSpecificRemainder` -> `add`, `hasRecipeRemainders` -> `hasRemainders`